app/controllers/course_reports_controller.rb

```
@@ -0,0 +1,151 @@
+    end
+  end
+
+  # function to retrieve final review due date from the db:
+  # Raises FinalDueDateNotReviewDeadline as error
+  def final_review_due_date_for(assignment)
+    final_due_date = assignment.due_dates.max_by(&:due_at)
+    return final_due_date.due_at if final_due_date&.deadline_type_id == DueDate::REVIEW_DEADLINE_TYPE_ID
+
+    # At this point of the project, all assignments are peer review assignments,
+    # so the final deadline is bound to be a review deadline, hence this guard
+    # 
```

RuboCop (1.86.1)

[convention] Trailing whitespace detected.

(Layout/TrailingWhitespace)